From 6600702ec9b7126a6b13c68d1fc48876c604ee53 Mon Sep 17 00:00:00 2001
From: andi34 <skate4life@gmx.de>
Date: Sat, 9 May 2015 10:48:52 -0700
Subject: [PATCH 1/2] updater: Allow devices to suppress BLKDISCARD

* On some devices TRIM is disabled for security reasons. Don't fail
  flashing the ROM because discard isn't possible in this case.

Change-Id: I044619c3e0b01a496d967ef136501d0190240ad4
---
 updater/Android.mk | 4 ++++
 updater/blockimg.c | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/updater/Android.mk b/updater/Android.mk
index 11e7bb8..ae3d703 100644
--- a/updater/Android.mk
+++ b/updater/Android.mk
@@ -29,6 +29,10 @@ LOCAL_STATIC_LIBRARIES += \
     libz
 endif
 
+ifeq ($(BOARD_SUPPRESS_EMMC_WIPE),true)
+    LOCAL_CFLAGS += -DSUPPRESS_EMMC_WIPE
+endif
+
 LOCAL_STATIC_LIBRARIES += $(TARGET_RECOVERY_UPDATER_LIBS) $(TARGET_RECOVERY_UPDATER_EXTRA_LIBS)
 LOCAL_STATIC_LIBRARIES += libapplypatch libedify libmtdutils libminzip libz
 LOCAL_STATIC_LIBRARIES += libmincrypt libbz
diff --git a/updater/blockimg.c b/updater/blockimg.c
index 6060ac2..9d7f0ea 100644
--- a/updater/blockimg.c
+++ b/updater/blockimg.c
@@ -729,10 +729,12 @@ Value* BlockImageUpdateFn(const char* name, State* state, int argc, Expr* argv[]
                     // len in bytes
                     range[1] = (tgt->pos[i*2+1] - tgt->pos[i*2]) * (uint64_t)BLOCKSIZE;
 
+#ifndef SUPPRESS_EMMC_WIPE
                     if (ioctl(fd, BLKDISCARD, &range) < 0) {
                         ErrorAbort(state, "    blkdiscard failed: %s\n", strerror(errno));
                         goto done;
                     }
+#endif
                 }
 
                 free(tgt);
-- 
2.7.4

